modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -574,6 +574,49 @@ OPENCV_HAL_IMPL_REDUCE_MINMAX_FUNC(v_reduce_min, std::min)
 For 32-bit integer and 32-bit floating point types. */
 OPENCV_HAL_IMPL_REDUCE_MINMAX_FUNC(v_reduce_max, std::max)
 
+/** @brief Count the 1 bits in the vector and return 4 values
+
+Scheme:
+@code
+{A1 A2 A3 ...} => popcount(A1)
+@endcode
+Any types but result will be in v_uint32x4*/
+template<typename _Tp, int n> inline v_uint32x4 v_popcount(const v_reg<_Tp, n>& a)
+{
+    static unsigned char popCountTable[] =
```

Comment:
> return the vector in v_uint8x16 rather than v_uint32x4

No, return type is the same (v_uint32x4).

I mean to change the type of table elements only: 
```
- static const unsigned char popCountTable[] =
+ static const int popCountTable[] =
```